pkg/services/preference/prefimpl/pref.go

```diff
@@ -18,6 +18,13 @@ type Service struct {
 }
 
 func ProvideService(db db.DB, cfg *setting.Cfg, features *featuremgmt.FeatureManager) pref.Service {
+	if cfg.IsFeatureToggleEnabled("newDBLibrary") {
+		return &Service{
+			store: &sqlxStore{
+				sess: db.GetSqlxSession(),
+			},
```

Comment:
need to set config and features, that is why there is the panic (you are returning before it).

Perhaps assign the service to a variable, and reassign the sess property to sqlx if feature flag enabled).